DevRel at PostHog Writeup

yakkomajuri opened the pull request:
A first stab at writing up about DevRel and particularly DX at PostHog.

This establishes some framework for us to look at this field, and is definitely meant to be iterated upon (hence I make a point of adding the last updated at date).

Please push back if something doesn't make sense.

It's a bit long, but this was a great exercise for myself as well to conceptualize this, and I spent some time thinking about longer-term hiring as per @jamesefhawkins' request at standup the other day.

One key takeaway for me was that there's quite a lot of stuff that I track but don't necessarily share out as much. I might spend some time setting up a dashboard with some metrics and start reporting during Monday's News session as well.

jamesefhawkins commented on contents/handbook/growth/developer-relations.mdx:
```diff
@@ -0,0 +1,232 @@
+
+As a developer-focused company offering a product that, in most cases, will require at least *some* developer involvement (and sometimes a lot of it!), Developer Relations is an important area to us.
+
+While we are not a dev tool, an appropriate PostHog setup requires that the customer, at a minimum, integrate a PostHog library into their codebase. Additionally, engineers may also step in to:
```
I would classify us as a devtool since we're adopted that way, even if we have a lot of UX

yakkomajuri replied:
Hmm I'd be against classifying us as a dev tool. We're not really focused on the engineering process - we're a company helping your Product process, even if we're developer-focused.

yakkomajuri replied:
Calling us a dev tool I believe would be misleading.

jamesefhawkins replied:
I disagree strongly here.

We are 100% a devtool in the broader market (VC world, and realistically the world of finding candidates to work for us - both see us this way). We target developers with our tool because we believe that analytics and product should be part of the development proecss (even if they aren't in many companies... that's a bad thing though).

I suspect longer term as we think harder about the dev process, product analytics will get closer to that (ie PRs over graphs, rolling out changes behind feature flags as a check in github, that sort of thing). Plugins are a step in this direction too.

The only reason we don't really feel like this is that we started as "an open source alternative to <product-manager-focussed-thing>"

It's kind of semantics, which I think is why we're disagreeing, but I see only downside to not thinking of us as a devtool from the purpose of this piece. We will need people that can code and want to work with developers to join.

seanpackham replied:
Historically, product analytics was not position at developers. We're flipping that, PostHog is positioned at developers to help them build more successful products.

yakkomajuri replied:
adjusted this

jamesefhawkins commented on contents/handbook/growth/developer-relations.mdx:
```diff
@@ -0,0 +1,232 @@
+|           User Interviews           |       Product, Engineering        |
+
+
+As we can see, a lot of the roles that are linked to "Developer Relations" are actually performed by "other teams" - and that's not only fine, it's encouraged. 
```
Style-wise, this isn't very concise "as we can see" = doesn't add anything

jamesefhawkins commented on contents/handbook/growth/developer-relations.mdx:
```diff
@@ -0,0 +1,232 @@
+
+For us, DX relates to improving the end-to-end experience for an engineer using (and/or building) our product. More on this later.
+
+##### Developer Advocacy and Evangelism
```
As I'm reading, I'm wondering if we do small teams #1067 , then we'd probably have dev relation people as part of the small teams (ie to really nail SDKs _or_ the website/docs), and perhaps dev advocacy would be part of a centralized growth function

yakkomajuri replied:
Yes I believe the Dev Advocates/Evangelists would probably end up falling under a centralized Growth or Marketing function. Unless we end up building a DevRel function too. But given our Marketing is led by a developer, that already borders on DevRel

jamesefhawkins commented on contents/handbook/growth/developer-relations.mdx:
```diff
@@ -0,0 +1,232 @@
+
+A dedicated copywriter will be able to do a better job of writing more top-of-funnel technical content aimed at acquiring new users.
+
+Additionally, given our short release cycles, the multiple facets of our product (Cloud, self-hosted, libraries, API, plugins, etc.), as well as our ambition for top-level design and feel for our Docs (token autofill, dark mode, search, etc.), getting our Docs to be world-class (as well as keeping them there) is definitely not a job for one single person - and potentially the job for a small team ...
```
I wonder if our website/docs team should look like:

* Graphic designer
* Product Manager (would _have_ to be an ex developer), similar to how Stripe approaches this, to think about the structure and features of our docs
* Developer to implement
* Lead Designer to get the UX right, perhaps temporarily until we've enough components we can move this role into the product teams

jamesefhawkins replied:
ie specifically _not_ including DevRel here, instead putting DevRel into the team that handles ie extensibility

yakkomajuri replied:
Yup - I think in the medium/long term that makes sense.

jamesefhawkins commented on contents/handbook/growth/developer-relations.mdx:
```diff
@@ -0,0 +1,232 @@
+On a more long-term view, we would also potentially benefit from having the following roles:
+
+- Developer Advocates/Evangelists/Educators: Acquistion-focused, external-facing, would take over some responsibilities from Marketing
+- Engineers for our libraries: Engineers who are also able to contribute to product but who would own certain key libraries (especially mobile libraries)
```
I strongly suspect we'll have a small team focussed on our libraries and should expand our efforts here

We have already decided to hire a mobile developer at least as a contractor initially to make sure our mobile libraries get updated and work more nicely

jamesefhawkins reviewed (changes requested):
Thank you for this, I've put in a few discussion points. I think #1067 is likely to happen (I've had positive feedback from everyone), so reading that first then tackling the points here would help

yakkomajuri commented:
Incorporated the feedback and removed the references to roles to hire into - it gave you @jamesefhawkins some insight into my thoughts but doesn't need to be "hard coded" on this page

yakkomajuri commented:
Anything else still needed here? @jamesefhawkins

yakkomajuri commented:
I added a little comment on Small Teams like we talked about @jamesefhawkins otherwise this should be good to go

jamesefhawkins commented:
Thanks, waiting for build then let's merge

yakkomajuri commented:
WIll merge - though I believe we're bringing someone in who will probably update this significantly

jamesefhawkins commented:
Gentle nudge - spring cleaning posthog.com repo!

yakkomajuri commented:
Oh, a month ago I enabled auto merge and I guess the check failed :( let me debug

yakkomajuri commented:
Merged #1431
